model/src/main/java/com/ge/predix/acs/model/PolicySet.java

```diff
@@ -19,17 +19,23 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+
 import io.swagger.annotations.ApiModel;
 import io.swagger.annotations.ApiModelProperty;
 
 /**
  * @author 212314537
  */
-@ApiModel(description = "A collection of access control policies evaluated in order. The first applicable policy"
-        + " determines the access control effect.")
+@ApiModel(
+        description = "A collection of access control policies evaluated in order. The first applicable policy"
+                + " determines the access control effect.")
 @SuppressWarnings("javadoc")
 public class PolicySet {
     private List<Policy> policies = new ArrayList<>();
+    @JsonProperty("obligationExpressions")
```

Comment:
Re: type - we need one which does not allow duplicates and preserves order.

Reply:
there is a validation for duplicated ids. I think order is only required on policies when you set the obligationexpressionsids you want to use.

Reply:
For obligationExpressions contained in a policy set we should use Set to avoid duplicates and remove the need to explicitly validate for duplicates.  Order is not important.

For obligationIds contained per each policy we should use LinkedHashSet to avoid duplicates and preserve the order.  

Obligations returned in the policy evaluation result should be returned in the same order as obligationIds in the matching policy.  Thus for Obligations contained in evaluation result we should also use LinkedHashSet to avoid duplicates and preserve the order.

Reply:
ok!

Reply:
After further discussion I agree with @sebastiantb that it might be better to explicitly validate policy set for duplicate obligationExpressions upon upsert and return an error.  Deserializing a json with duplicate obligationExpressions into LinkedHashSet might produce non-intended obligation content.  

The same argument goes for obligationIds.  Deserializing into LinkedHashSet might produce non-intended order by silently replacing a duplicate ids.  For example, 1, 2, 3, 1 will be deserialized into 1,2,3.  What is intension was to keep 1 last?  It's better to explicitly validate for duplicates upon policy set upsert and return an error.

In that case using List is sufficient.  If we agree I will update the design to reflect that.